src/query/exist_query.rs

```diff
@@ -0,0 +1,306 @@
+use core::fmt::Debug;
+
+use columnar::{ColumnIndex, DynamicColumn};
+
+use super::{ConstScorer, EmptyScorer};
+use crate::core::SegmentReader;
+use crate::docset::{DocSet, TERMINATED};
+use crate::query::explanation::does_not_match;
+use crate::query::{EnableScoring, Explanation, Query, Scorer, Weight};
+use crate::{DocId, Score, TantivyError};
+
+/// Query that matches all documents with a non-null value in the specified field.
+///
+/// All of the matched documents get the score 1.0.
+#[derive(Clone, Debug)]
+pub struct ExistsQuery {
+    field_name: String,
+}
+
+impl ExistsQuery {
+    #![allow(dead_code)]
```

Comment:
```suggestion
```

Reply:
Why would we allow dead code? The compiler is informing you that there is no way to call this method.

Reply:
I am not sure I follow. It is not hooked up to grammar at the moment, so the only users of this will be quickwit's ExistsQuery for now.

Reply:
Oh, yeah, I keep forgetting that `pub use` thing.